### PR TITLE
fix: customerIo: remove anonymousId as fallback in identify call

### PR DESCRIPTION
**Pull request by anantjain45823**
## PR Description
Resolves INT-491
Removing anonymousId as fallback for identify call in customerIo to maintain parity between cloud and device mode and not create duplicate user

## Notion ticket

[Ticket link
](https://linear.app/rudderstack/issue/INT-491/remove-anonid-as-fallback-from-device-mode)
## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/integrations/CustomerIO/browser.js [hunk 1]
```diff
@@ -30,10 +30,11 @@ class CustomerIO {
...
   identify(rudderElement) {
     logger.debug('in Customer IO identify');
-    const userId = rudderElement.message.userId
-      ? rudderElement.message.userId
-      : rudderElement.message.anonymousId;
-    const traits = rudderElement.message.context.traits ? rudderElement.message.context.traits : {};
+    const { userId, context } = rudderElement.message;
+    const { traits } = context || {};
+    if(!userId){
+      logger.error('userId is required for Identify call.');
```
Comment (sanpj2292) [inline src/integrations/CustomerIO/browser.js:36]: We don't need to return here ? 🤔

Comment (anantjain45823) [reply src/integrations/CustomerIO/browser.js:36]: Yes we do. Added now

Comment (sanpj2292) [reply src/integrations/CustomerIO/browser.js:36]: Will userId be only available in `message.userId` or should we consider something others as well (like how we do in [GenericFieldMapping.json](https://github.com/rudderlabs/rudder-transformer/blob/develop/src/v0/util/data/GenericFieldMapping.json#L12-L26))

Comment (anantjain45823) [reply src/integrations/CustomerIO/browser.js:36]: In most of the integrations we search for it at root level only but yes w can use GenericFieldMapping as well. Shall I make the change

Comment (sanpj2292) [reply src/integrations/CustomerIO/browser.js:36]: It is fine if it is being done for most of the integrations

**Review by sanpj2292 — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**